daemon/checkpoint.go

```diff
@@ -38,6 +38,11 @@ func (daemon *Daemon) CheckpointCreate(name string, config types.CheckpointCreat
 		return fmt.Errorf("Invalid checkpoint ID (%s), only %s are allowed", config.CheckpointID, validCheckpointNameChars)
 	}
 
+	checkDir := filepath.Join(daemon.repository, container.ID, "checkpoints", config.CheckpointID)
+	if stat, err := os.Stat(checkDir); err == nil && stat.IsDir() {
+		return fmt.Errorf(" checkpoint with name %s already exists for container %s", config.CheckpointID, container.Name)
+	}
+
 	err = daemon.containerd.CreateCheckpoint(container.ID, config.CheckpointID, checkpointDir, config.Exit)
 	if err != nil {
```

Comment:
@thaJeztah,
Thanks for the review, I will update the PR accordingly.

I observed that when I created checkpoint with custom directory(--checkpoint-dir ) option,  "docker checkpoint ls" command does not show any output.

When checkpoint is create without custom checkpoint-dir, checkpoint is created under /var/lib/docker/container/<containerID>/checkpoints/
 
How can we find dependancy between checkpoints and containers when multiple checkpoints created for different containers in same directory using --checkpoint-dir ?